### PR TITLE
KAFKA-12216; Improve flaky test reporting

### PR DESCRIPTION
**Pull request by dajac**
This patch adds the `reports.junitXml.mergeReruns` setting to gradle build. From the [doc](https://docs.gradle.org/current/userguide/java_testing.html#mergereruns):

> When mergeReruns is enabled, if a test fails but is then retried and succeeds, its failures will be recorded as `<flakyFailure>` instead of `<failure>`, within one `<testcase>`. This is effectively the reporting produced by the [surefire plugin of Apache Maven™](https://maven.apache.org/components/surefire/maven-surefire-plugin/examples/rerun-failing-tests.html) when enabling reruns. If your CI server understands this format, it will indicate that the test was flaky. If it does not, it will indicate that the test succeeded as it will ignore the `<flakyFailure>` information. If the test does not succeed (i.e. it fails for every retry), it will be indicated as having failed whether your tool understands this format or not.

The patch also increases the `maxTestRetryFailures` to `50`.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java [hunk 1]
```diff
@@ -239,10 +240,14 @@ public static String safeUniqueTestName(final Class<?> testClass, final TestName
...
-        final String displayName = testInfo.getDisplayName();
         final String methodName = testInfo.getTestMethod().map(Method::getName).orElse("unknownMethodName");
-        final String testName = displayName.contains(methodName) ? methodName : methodName + displayName;
-        return safeUniqueTestName(testClass, testName);
+        // Generate a random uuid without an `-`. The `-` is used in Streams' thread name as
+        // a separator and some tests rely on this.
+        String randomUuid;
+        do {
+            randomUuid = Uuid.randomUuid().toString();
+        } while (randomUuid.contains("-"));
```
Comment (cadonna) [inline streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java]: Why not just replacing `-` with a different character?

Comment (dajac) [reply streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java]: Addressed it in the original PR: https://github.com/apache/kafka/pull/14862.

**Comment by dajac** [comment]
We reverted https://github.com/apache/kafka/pull/14961 because it created flakiness in Streams' tests. We should bring it back before we can continue with this one.

**Comment by dajac** [comment]
Rebased to include https://github.com/apache/kafka/commit/87e3cbe4daa084964596d3f6f79041d85548426b.

**Comment by dajac** [comment]
I need to update the PR to only apply this while building PRs.

**Pull request closed without merging**